### ISSUE TITLE
YANOM-Note-O-Matic convert problem

### ISSUE THREAD

**Issue by pousejo** [kevindurston21/YANOM-Note-O-Matic#63]
YANOM-Note-O-Matic doesn't convert nix-files to html on MacOS

- input: nsx (files with no encrypted notes)
- output: html

Note-files in html

4 screenshots as attachments
<img width="582" alt="Schermafbeelding 2021-07-22 om 16 08 29" src="https://user-images.githubusercontent.com/87821288/126666312-c21e97a9-63cc-4bc5-b4ad-cbaef1428d47.png">
<img width="667" alt="Schermafbeelding 2021-07-22 om 16 11 35" src="https://user-images.githubusercontent.com/87821288/126666316-1a770f43-cc29-462c-b6af-8f19f384844f.png">
<img width="647" alt="Schermafbeelding 2021-07-22 om 16 14 45" src="https://user-images.githubusercontent.com/87821288/126666320-4d11baa7-879b-49b9-9970-ced9d8c9df28.png">
<img width="647" alt="Schermafbeelding 2021-07-22 om 16 15 17" src="https://user-images.githubusercontent.com/87821288/126666322-fa7274c8-987b-4d6d-83da-eda44866cbdc.png">

MacOS 11.4 (Big Sur)
Terminal + YANOM-Note-O-Matic 1.3.0

Result:
no error in Terminal
all errors in the log-files
no files in the notes-folder

**Comment by kevindurston21** [comment]
Fixed in 1.3.1

Thanks pousejo,

I found it was caused by a '.' in the notebook names being recognised when creating the folder but then later when used to create the path for the notes to be saved to it was truncated at the dot, so in your example it was trying to save the notes to a folder '230' which did not exist

I have fixed this in 1.3.1 - I have built a mac release for this so far and I will upload that in the next few minutes

![image](https://user-images.githubusercontent.com/43143552/126718997-42e53e91-bb8e-4ad8-aeac-6d32e08f26fb.png)


Please give this version a try.  Let me know if it works OK.

Cheers

Kevin

**Comment by pousejo** [comment]
Hi Kevin

There is good news and bad news.

The good news!
I converted with Yanom 1.3.1 a note block with 4 notes. The note block was named "050.00 - handleidingen". One of the notes was "313.67 - Televisie : Philips 32pfl6007h_12". The others have no numeric code. The conversion was fine.
I imported it to Apple Notes. Spaces in the title of the note block were gone. Besides that all goes well, but the attachments were not embedded. So I have manually to drop them in the note. The links to the attachments are correct.

<img width="585" alt="Schermafbeelding 2021-07-23 om 10 43 33" src="https://user-images.githubusercontent.com/87821288/126758296-05c3bb18-1bcb-4a97-9132-68c24e859318.png">
<img width="664" alt="Schermafbeelding 2021-07-23 om 10 45 01" src="https://user-images.githubusercontent.com/87821288/126758300-2acd5236-cd77-4f57-865c-7d50f487f818.png">


The bad news!
I wanted to convert the note block "230.00 - Geschiedenis" with 549 notes. Nothing works. I changed the block note name without numbers, dots, ... but nothing.
I have other note blocks with more ar less notes and tried, but nothing works.
The error log is always empty.
I can only guess, but i didn't know, that there is een restriction by conversion on the nummers of notes in a note block or on the length of the note or on the mb.

<img width="587" alt="Schermafbeelding 2021-07-23 om 11 08 29" src="https://user-images.githubusercontent.com/87821288/126761043-ebaafc5b-125d-4912-843c-1cfd7014ab17.png">
[normal.log](https://github.com/kevindurston21/YANOM-Note-O-Matic/files/6867734/normal.log)


Cheers
Jozef

**Comment by kevindurston21** [comment]
Hi Jozef,

First thing..  The `TypeError' failure for the attachments - this looks like the same issue Elmer had with his encrypted notes.  I think there is a note coming out of note station with it's attachment field set to 'null'.   

I have a fix for this and was going to put that out with 1.4.0, but I'm having issues with windows for 1.4.0 (always a problem with the windows build!).  

I have started making a 1.3.2 which will hopefully help with this `null` problem.  It will see null, not crash and the note export but will not have any attachments.  The log file will have a `WARNING` line when it does this so you can see if it is happening. 

```python
f'Note - {self._title} - Has Null set for attachments. '
 f'There may be a sync issues between desktop and web version of Note Station.')
```
I think the root cause is something in Note Station where files do not sync between desktop and server but I can not work out what is going on exactly.  You may find attachments missing but as you are having to drop them into notes manually it should work enough for you to get a full set of notes out, but as the actual attachments are not exported you will need to save them manually out of note station.

If the issue is syncing then perhaps try an export from desktop or web version (which ever one you did not use for the above problem) and see if that will actually generate an nsx file with out the 'null' value.  The testing I did it always worked from at least one of them.  I'd like to do more testing but my desktop and server will no longer sync at all now!  I did update to the latest version of both and I think that has stopped all syncing on my system.

1.3.2 should be out for mac in the next hour or so. If I have problems with Windows I'll just do the mac version like I did for 1.3.1 and then work on the windows one later.

Second Thing.  Apple notes, I see what you mean.  I'm afraid no solution there as you need to embed the attachment files in the note as an actual file for portability across mac / ipad / phone.  The hyper links do work to the local file copy in the attachments folder but not much help with cross device portability :-)

![image](https://user-images.githubusercontent.com/43143552/126768066-bb709271-e246-45ea-ba3a-4b92e33e7ff4.png)

Cheers

Kevin

**Comment by kevindurston21** [comment]
Hi

1.3.2 is now uploaded

Kevin

FIXED in v1.3.2
